### DISCUSSION TITLE
Blit Surface causes window to crash

### DISCUSSION THREAD

**Discussion by sujaldev** [browserengineering/book#328] [Q&A] [answered]
When I create a window approximately greater than 800x800, calling `SDL_BlitSurface` freezes the window with cpu usage spiking upto 100% and then exits with the following message:

> Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)

I am not so familiar with lower level languages but from what a quick google search tells me this is a segmentation fault, any tips on how to fix this?

[<sup>Context</sup>](https://browser.engineering/visual-effects.html#sdl-creates-the-window)

**Comment by chrishtr** [comment, marked as answer]
Interesting. SIGSEGV means that the program tried to read an illegal place memory, due to an error coding, usually a null dereference or use-after-free.

I tried WIDTH, HEIGHT = 1000, 1000 just now and it worked fine on my machine.

My guess is your code has a bug that the surfaces you're copy to and from via SDL_BlitSurface have different dimensions.

**Reply by sujaldev** [reply]
Turns out the dimensions weren't the problem but the use-after-free problem. I was storing the pixel data returned by `tobytes` in a local variable which was being garbage collected when it was larger by python and `SDL_CreateRGBSurfaceFrom` doesn't copy that data but keeps a pointer to the same variable.
Thanks for your help anyways!!

**Reply by chrishtr** [reply]
Great, glad you found the bug.

**Reply by chrishtr** [reply]
I [fixed the text](https://github.com/browserengineering/book/pull/332) also.
